### ISSUE TITLE
Case where [no-invalid-position-declaration] rule is not working

### ISSUE THREAD

**Issue by jimblue** [stylelint/stylelint#9119]
### What minimal example or steps are needed to reproduce the bug?

Getting an [**no-invalid-position-declaration error**](https://stylelint.io/user-guide/rules/no-invalid-position-declaration/) for following mixin:

```scss
@mixin text-smart-justification($justify: true) {
  max-inline-size: var(--text-max-inline-size);
  text-align: left;

  @if $justify {
    text-align: justify;
    text-align-last: left;
    hyphens: auto;
    hyphenate-limit-chars: 6 3 3;
    hyphenate-limit-lines: 2;

    @supports (text-wrap: pretty) {
      text-wrap: pretty; // stylelint-disable-line no-invalid-position-declaration
    }
  }
}
```

<img width="717" height="300" alt="Image" src="https://github.com/user-attachments/assets/4ffd4da0-ec54-492f-8059-41505fe0fce0" />

### What minimal configuration is needed to reproduce the bug?

```json
{
  "no-invalid-position-declaration": true
}
```

### How did you run Stylelint?

`npx stylelint --allow-empty-input --fix --report-needless-disables '**/*.{css,scss}'`

### Which Stylelint-related dependencies are you using?

- stylelint `17.4.0`
- stylelint-config-recommended-scss `17.0.0`
- stylelint-order `7.0.1`

### What did you expect to happen?

The code is valid and should not output an error.

### What actually happened?

**Unexpected invalid position declaration** ([no-invalid-position-declaration](https://stylelint.io/user-guide/rules/no-invalid-position-declaration))

**Comment by ragini-pandey** [comment]
When a declaration's parent is a nesting-supported at-rule (e.g., `@supports`), the rule walks up the tree using findNodeUpToRoot looking for a parent rule (selector) or an ignored at-rule. It walked past non-nesting-supported at-rules like `@if` and `@mixin` without recognizing them as valid contexts, even though declarations directly inside those at-rules are already allowed by the rule.

**Comment by romainmenke** [comment]
Hi @jimblue,

You can ignore non-standard atrules with the `ignoreAtRules` rule option:

```js
{
     "no-invalid-position-declaration": [true, { ignoreAtRules: ['if', 'mixin'] }]
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBbAlgDzQOwAQDMIJdgAdPXSAGwgCd5daYoBuc83XJAZwFcAHfnQAu3XAApqdeEygBKEh06UINeo2ZsKAX3YUkafLgAkAK17dhhgJ6KKnKetlalnHgKG1RExzOYKye2VfDVZXXF0dcm0QABoQfDQqGAA5AEMUOEQYDAz+ZIA6MG5uOPAIbESAcwQQHM9hXFh8NN4qRsDOWjaYbgZO5VxSEGwIAFocADc0qjQoMaFuNCsKsdgwKjTaNJXsYYYAbWFumFiSAAK0KtGmAEFhACUevtwDgHJDN7O39CxsN4AuhEAUpIpEypBKlcAGJ0FA7WqWazJWbYYRFCrVAooUyleKwfilRADYZIlE4YT7IYgTbCXqUuJKUnCZEwVHCMaQ6pjSxpbBQLZQKnDWn04bRcggbRAA)

----

`@mixin` happens to be currently going through standardization and we can add support for it in this rule without requiring further config, but we can't do the same for `@if`.

**Comment by jimblue** [comment]
Hey @romainmenke,

Thank you so much for pointing me to this solution!
